fix(checkbox): checked state forced color contrast

dancormier opened the pull request:
[STACKS-599](https://stackoverflow.atlassian.net/browse/STACKS-599)

---

This PR fixes the lack of contrast in the checked checkbox component when the user has enabled forced color mode.

### Before

![stackoverflow design_product_components_checkbox_](https://github.com/StackExchange/Stacks/assets/647177/9743d71d-bb79-47aa-a226-9990897f29fd)

### After

![localhost_8080_product_components_checkbox_](https://github.com/StackExchange/Stacks/assets/647177/aca15531-6745-49cf-8342-9b136e5a4704)

dancormier committed:
fix(checkbox): checked state forced color contrast

[STACKS-599](https://stackoverflow.atlassian.net/browse/STACKS-599)

dancormier committed:
We don't need those `!important`s

dancormier committed:
I was wrong and we do need those `!important`s

giamir reviewed (changes requested):
Based on my testing in WHCM (Desert mode) we can achieve what we want without having to redefine the bg-image
This is sufficient:
```
   @media (forced-colors: active) {
        &:checked {
            --_ch-bg: ButtonText !important;
        }
        &:indeterminate {
            --_ch-bg: ButtonText !important;
        }
    }
```

I am happy to merge the simplified version above in. 

**Sidenote**
While I was in Windows I took some time to explore our DS in WHCM and I could see several components that are not rendering well (including toggles, tags, navigation not showing selected items, etc...).  We should probably create a rough inventory of those issues as part of the upcoming spike story.

dancormier commented:
> Based on my testing in WHCM (Desert mode) we can achieve what we want without having to redefine the bg-image This is sufficient:
> 
> ```
>    @media (forced-colors: active) {
>         &:checked {
>             --_ch-bg: ButtonText !important;
>         }
>         &:indeterminate {
>             --_ch-bg: ButtonText !important;
>         }
>     }
> ```
> 
> I am happy to merge the simplified version above in.

Without the background image redefinition, we'll have contrast issues if Stacks dark mode is enabled in addition to forced color mode.

![image](https://github.com/StackExchange/Stacks/assets/647177/6b6624ed-5861-4432-9d7f-8f365e6e7cbf)

I understand this would be an unusual circumstance but I figured it was a good idea to account for all possible cases. What do you think @giamir?

> **Sidenote** While I was in Windows I took some time to explore our DS in WHCM and I could see several components that are not rendering well (including toggles, tags, navigation not showing selected items, etc...). We should probably create a rough inventory of those issues as part of the upcoming spike story.

Yep, there's definitely room for improvement across our components. The [ticket](https://stackoverflow.atlassian.net/browse/STACKS-559) has "A rough inventory of user forced color issues in our components (e.g. toggles, tags, etc…) is added to the document" in the success criteria so I'll make sure to include a list of components (assuming I perform the spike sprint).

dancormier committed:
Only alter --_ch-bg in forced color mode

From suggestion made in https://github.com/StackExchange/Stacks/pull/1699#pullrequestreview-1979464312
